pkg/block/metadata/hash.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package metadata
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+// HashFunc indicates what type of hash it is.
+type HashFunc string
+
+const (
+	// SHA256Func shows that SHA256 has been used to generate the hash.
+	SHA256Func HashFunc = "SHA256"
+	// NoneFunc shows that hashes should not be added. Used internally.
+	NoneFunc HashFunc = ""
+)
+
+// ObjectHash stores the hash of an object in the object storage.
+type ObjectHash struct {
+	Func  HashFunc
+	Value string
+}
+
+// Equal returns true if two hashes are equal.
+func (oh *ObjectHash) Equal(other *ObjectHash) bool {
+	return oh.Value == other.Value
+}
+
+// CalculateHash calculates the hash of the given type.
+func CalculateHash(p string, hf HashFunc) (ObjectHash, error) {
+	switch hf {
+	case SHA256Func:
+		f, err := os.Open(p)
+		if err != nil {
+			return ObjectHash{}, errors.Wrap(err, "opening file")
+		}
+		defer f.Close()
```

Comment:
I guess here we could apply the Coding-style guide?
https://github.com/thanos-io/thanos/blob/master/docs/contributing/coding-style-guide.md#defers-dont-forget-to-check-returned-errors